cmd/api/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"flag"
+	"time"
+
+	"github.com/airbrake/gobrake"
+	"github.com/gin-gonic/gin"
+)
+
+var env = flag.String("env", "development", "environment, e.g. development or production")
+var host = flag.String("host", "", "host")
+var projectId = flag.Int64("project_id", 0, "project ID")
+var projectKey = flag.String("project_key", "", "project key")
+
+var api *gin.Engine
+var pathMap map[string]string
+var notifier *gobrake.Notifier
+
+func main() {
+	flag.Parse()
+
+	notifier = gobrake.NewNotifierWithOptions(&gobrake.NotifierOptions{
+		ProjectId:   *projectId,
+		ProjectKey:  *projectKey,
+		Host:        *host,
+		Environment: *env,
+	})
+
+	api = gin.Default()
+	api.Use(reportRouteStats)
+	api.GET("/hello/:name", hello)
+	api.GET("/ping", ping)
+	api.Run(":8080")
+}
+
+func reportRouteStats(c *gin.Context) {
+	var path string
+	pathMap = make(map[string]string)
+	if in, ok := pathMap[c.HandlerName()]; ok {
```

Comment:
this is not thread safe - several goroutines can read and write to the map at the same time - https://golang.org/doc/faq#atomic_maps